### ISSUE TITLE
prefab-1.3.1.0.jar - java.lang.NoClassDefFoundError: Lcom/wuest/prefab/proxy/CommonProxy;

### ISSUE THREAD

**Issue by RussellHaley** [Brian-Wuest/MC-Prefab#186]
1. What version of Minecraft, Forge and Mod are you running?
    1. Minecraft Version: 1.12.2
    1. Forge Version: 14.23.5.2838
    1. Mod Version: 1.3.1.0
1. What type of issue is this? Choose one of the following:
    1. Crash - java.lang.NoClassDefFoundError: Lcom/wuest/prefab/proxy/CommonProxy;

Looks like this is a brand new release (release only a few hours ago)? I'll try the older version. Thanks for all your great work. I wish I had time to work on mods. 

Crash report: 

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  MicdoodlePlugin (MicdoodleCore-1.12.2-4.0.2.280.jar)
  FMLModule (MDXLib-1.12.2.jar)
Contact their authors BEFORE contacting forge

// Hi. I'm Minecraft, and I'm a crashaholic.

Time: 6/18/21 6:11 PM
Description: Initializing game

java.lang.NoClassDefFoundError: Lcom/wuest/prefab/proxy/CommonProxy;
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
	at java.lang.Class.getDeclaredField(Class.java:2068)
	at net.minecraftforge.fml.common.ProxyInjector.inject(ProxyInjector.java:70)
	at net.minecraftforge.fml.common.FMLModContainer.constructMod(FMLModContainer.java:611)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
	at net.minecraftforge.fml.common.Loader.loadMods(Loader.java:593)
	at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:232)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:467)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:378)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
Caused by: java.lang.ClassNotFoundException: com.wuest.prefab.proxy.CommonProxy
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 41 more
Caused by: java.lang.NullPointerException


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
	at java.lang.Class.getDeclaredField(Class.java:2068)
	at net.minecraftforge.fml.common.ProxyInjector.inject(ProxyInjector.java:70)
	at net.minecraftforge.fml.common.FMLModContainer.constructMod(FMLModContainer.java:611)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
	at net.minecraftforge.fml.common.Loader.loadMods(Loader.java:593)
	at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:232)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:467)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:378)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 465929896 bytes (444 MB) / 1073741824 bytes (1024 MB) up to 4294967296 bytes (4096 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=100M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.42 Powered by Forge 14.23.5.2838 Optifine OptiFine_1.12.2_HD_U_E3 20 mods loaded, 20 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State | ID                  | Version      | Source                                    | Signature                                |
	|:----- |:------------------- |:------------ |:----------------------------------------- |:---------------------------------------- |
	| LC    | minecraft           | 1.12.2       | minecraft.jar                             | None                                     |
	| LC    | mcp                 | 9.42         | minecraft.jar                             | None                                     |
	| LC    | FML                 | 8.0.99.99    | forge-1.12.2-14.23.5.2838.jar             | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LC    | forge               | 14.23.5.2838 | forge-1.12.2-14.23.5.2838.jar             | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LC    | micdoodlecore       |              | minecraft.jar                             | None                                     |
	| LC    | mdxlib              | 3.0.0.21     | MDXLib-1.12.2.jar                         | None                                     |
	| LC    | avp                 | 6.0.0.26     | Aliens-vs-Predator-Mod-1.12.2.jar         | None                                     |
	| LC    | blocklings          | 1.0          | Blocklings+6.0.1_b+-+1.12.2.jar           | None                                     |
	| LC    | customspawner       | 3.11.4       | Custom-Mob-Spawner-Mod-1.12.2.jar         | None                                     |
	| LC    | cfm                 | 6.3.1        | furniture-6.3.1-1.12.2.jar                | None                                     |
	| LC    | galacticraftcore    | 4.0.2.280    | GalacticraftCore-1.12.2-4.0.2.280.jar     | None                                     |
	| LC    | galacticraftplanets | 4.0.2.280    | Galacticraft-Planets-1.12.2-4.0.2.280.jar | None                                     |
	| LC    | ichunutil           | 7.1.4        | iChun-Util-Mod-1.12.2.jar                 | None                                     |
	| LC    | lucky               | 7.6.0        | luckyblock-1.12.2-7.6.0.zip               | None                                     |
	| LC    | mocreatures         | 12.0.5       | Mo-Creatures-Mod-1.12.2.jar               | None                                     |
	| LC    | morph               | 7.1.2        | Morph-1.12.2.jar                          | None                                     |
	| LC    | overloaded          | 0.0.59       | Overloaded-1.12.2-0.0.59.jar              | None                                     |
	| LC    | portalgun           | 7.0.2        | Portal-Gun-Mod-1.12.2.jar                 | None                                     |
	| L     | prefab              | @VERSION@    | prefab-1.3.1.0.jar                        | None                                     |
	| L     | tm                  | 1.0          | Transformers+Unlimited+1.1.2.jar          | None                                     |

	Loaded coremods (and transformers): 
MicdoodlePlugin (MicdoodleCore-1.12.2-4.0.2.280.jar)
  micdoodle8.mods.miccore.MicdoodleTransformer
FMLModule (MDXLib-1.12.2.jar)
  com.asx.mdx.AccessTransformer
	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 456.71' Renderer: 'GeForce GTX 660/PCIe/SSE2'
	Launched Version: 1.12.2-forge1.12.2-14.23.5.2838
	LWJGL: 2.9.4
	OpenGL: GeForce GTX 660/PCIe/SSE2 GL version 4.6.0 NVIDIA 456.71, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: vanilla (incompatible), file/Siren Head resource pack.zip
	Current Language: English (Canada)
	Profiler Position: N/A (disabled)
	CPU: 8x Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz
```

**Comment by Dqu1J** [comment]
Hello! When does this issue occur and how to reproduce it? Thank you.

**Comment by RussellHaley** [comment]
Hi, minecraft never loads. I press start in the launcher and it crashes shortly after I see the forge screen. I just checked and prefab-1.3.0.26.jar works for me (minecraft loaded, can see the build gui, haven't tried building yet).

Here is the list of mods (my kids' computer, not my fault!):

PS C:\Users\anika\AppData\Roaming\.minecraft> ls mods | Select-Object Name

Name
----
Aliens-vs-Predator-Mod-1.12.2.jar
Blocklings+6.0.1_b+-+1.12.2.jar
Custom-Mob-Spawner-Mod-1.12.2.jar
furniture-6.3.1-1.12.2.jar
Galacticraft-Planets-1.12.2-4.0.2.280.jar
GalacticraftCore-1.12.2-4.0.2.280.jar
iChun-Util-Mod-1.12.2.jar
luckyblock-1.12.2-7.6.0.zip
MDXLib-1.12.2.jar
MicdoodleCore-1.12.2-4.0.2.280.jar
Mo-Creatures-Mod-1.12.2.jar
Morph-1.12.2.jar
OptiFine_1.12.2_HD_U_E3.jar
OptiFine_1.16.1_HD_U_G2.jar
Overloaded-1.12.2-0.0.59.jar
Portal-Gun-Mod-1.12.2.jar
prefab-1.3.0.26.jar
Transformers+Unlimited+1.1.2.jar

**Comment by RussellHaley** [comment]
(Son wants computer back now that the mod works. Kids don't understand, lol. I'll check back in a couple of hours)

**Comment by Dqu1J** [comment]
I'm able to reproduce this issue.
It is not an mod incompatability issue as it occurs with only Prefab installed.

Forge version: 14.23.5.2854

**Comment by Brian-Wuest** [comment]
Yikes, I changed the casing of all of the folders to match what I am doing in 1.6.5 but somehow the built mod file didn't respect that.

Archiving version and building a new one

**Comment by Brian-Wuest** [comment]
This has been fixed with 1.3.1.1, I made sure that it works on curseforge this time....very sorry about that!

I have archived the non-working mod file and released a new one. It should be available on CurseForge soon.

Again; sorry about this.
